File: 2099-find-subsequence-of-length-k-with-the-largest-sum/2099-find-subsequence-of-length-k-with-the-largest-sum.java
class Solution {
    public int[] maxSubsequence(int[] nums, int k) {
        int[] temp = new int[nums.length];
        int[] result = new int[k];
        temp = nums.clone();
        Arrays.sort(temp);
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = temp.length-1; i > temp.length-1-k; i--)
        {
            if(!map.containsKey(temp[i]))
                map.put(temp[i], 1);
            else
                map.compute(temp[i], (key, v) -> v+=1);
        }
        int i = 0;
        for(int num: nums)
        {
            if(map.containsKey(num) && map.get(num) > 0)
            {
                result[i] = num;
                map.compute(num, (key, v) -> v-=1);
                i++;
            }
        }
        return result;
    }
}

File: 2099-find-subsequence-of-length-k-with-the-largest-sum/2099-find-subsequence-of-length-k-with-the-largest-sum.cpp
class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k) {
        
        vector<pair<int,int>> vect;
        int n=nums.size();
        for(int i=0; i<n;i++)
        {
            vect.push_back({nums[i],i});
        }
        sort(vect.begin(), vect.end());
        long long int sum=0;
        vector <pair<int, int>> vect2;
        for(int i=n-1;i>=n-k;i--)
        {
            vect2.push_back({vect[i].second, vect[i].first});
        }
        sort(vect2.begin(), vect2.end());
        vector<int>vect3;
        for(int i=0; i<vect2.size();i++)
        {
            vect3.push_back(vect2[i].second);
        }
        return vect3;
    }
};